src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -769,6 +771,205 @@ protected bool IsAlive
+            // For Beam systems we get each indivdual Beam as the childs
+            List<ElementId> beamSystemComponentIds = beamSystemElement.GetBeamIds().ToList();
+
+            if (beamSystemComponentIds.Count == 0)
+                throw new InvalidOperationException(Properties.Resources.NoChildElements);
+
+            List<Element> beamSystemComponentElements = beamSystemComponentIds
+                .Select(id => Document.GetElement(id).ToDSType(true))
+                .ToList();
+            return beamSystemComponentElements;
+        }
+
+        private static List<Element> GetChildElementsFromStairs(Autodesk.Revit.DB.Element element)
+        {
+            var stairElement = element as Autodesk.Revit.DB.Architecture.Stairs;
+            List<ElementId> stairComponentIds = new List<ElementId>();
+
+            // For stairs we get Landings, Runs and Supports as the child Elements,
+            // using the following methods.
+            stairComponentIds.AddRange(stairElement.GetStairsLandings().ToList());
+            stairComponentIds.AddRange(stairElement.GetStairsRuns().ToList());
+            stairComponentIds.AddRange(stairElement.GetStairsSupports().ToList());
+            if (stairComponentIds.Count == 0)
+                throw new InvalidOperationException(Properties.Resources.NoChildElements);
+
+            List<Element> stairComponentElements = stairComponentIds.Select(id => Document.GetElement(id).ToDSType(true))
+                                                                     .ToList();
+            return stairComponentElements;
+        }
+
+        /// <summary>
+        /// Gets the parent element of the Element.
+        /// </summary>
+        /// <returns>Parent Element</returns>
+        public Element GetParentElement()
+        {
+            return GetParentElementFromElement(this.InternalElement);
+        }
+
+        private Element GetParentElementFromElement(Autodesk.Revit.DB.Element element)
+        {
+            Autodesk.Revit.DB.Element parent;
+            BuiltInCategory builtInCategory = (BuiltInCategory)System.Enum.Parse(typeof(BuiltInCategory),
+                                                                                 element.Category.Id.ToString());
+
+            // By default we use the SuperComponent on the elements FamilyInstance to get the Parent Element, 
+            // for now the node also handles special cases including Stairs, StructuralFramingSystems and Railings 
+            switch (builtInCategory)
+            {
+                case BuiltInCategory.OST_StairsLandings:
+                    parent = GetParentComponentFromStairsLandings(element);
+                    break;
+
+                case BuiltInCategory.OST_StructuralFraming:
+                    parent = GetParentElementFromStructuralFraming(element);
+                    break;
+
+                case BuiltInCategory.OST_Railings:
```

Comment:
From GetChildElements, OST_Railings can get HandRails and TopRail as its child elements. But here shows, it can't get the parent element if the element is hand rails or top rails. This makes these two methods irreversible. I think it's better to add these two cases: 
case BuiltInCategory.OST_RailingHandRail:
case BuiltInCategory.OST_RailingTopRail:
What do you think of this?